File: saizad-mvvm/src/main/java/com/saizad/mvvm/model/IdModel.java
package com.saizad.mvvm.model;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.CallSuper;

import com.google.gson.annotations.SerializedName;

public class IdModel implements Parcelable {
    @SerializedName("id")
    public int id;

    public IdModel(int id) {
        this.id = id;
    }

    protected IdModel(){}

    protected IdModel(Parcel in) {
        id = in.readInt();
    }

    @Override
    @CallSuper
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(id);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<IdModel> CREATOR = new Creator<IdModel>() {
        @Override
        public IdModel createFromParcel(Parcel in) {
            return new IdModel(in);
        }

        @Override
        public IdModel[] newArray(int size) {
            return new IdModel[size];
        }
    };
}

File: saizad-mvvm/src/main/java/com/saizad/mvvm/model/IdModel.kt
package com.saizad.mvvm.model

import android.os.Parcel
import android.os.Parcelable

open class IdModel(val id: Int = 0) : Parcelable {

    constructor(parcel: Parcel) : this(parcel.readInt())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(id)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<IdModel> {
        override fun createFromParcel(parcel: Parcel): IdModel {
            return IdModel(parcel)
        }

        override fun newArray(size: Int): Array<IdModel?> {
            return arrayOfNulls(size)
        }
    }
}